src/test/java/com/floragunn/searchguard/WildcardMatcherTest.java

```diff
@@ -0,0 +1,141 @@
+package com.floragunn.searchguard.support;
+
+import com.floragunn.searchguard.support.WildcardMatcher;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class WildcardMatcherTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  @Rule public Timeout globalTimeout = new Timeout(10000);
+
+  @Test
+  public void containsWildcardInputNotNullOutputFalse() {
```

Comment:
I think we can work on the test names to convey better which case of the source code is covered. This would help the reviewer to understand why this test is there.